src/ports/nfts/component.ts

```diff
@@ -89,6 +98,27 @@ export function createNFTComponent<T extends { id: string }>(options: {
       )
     }
 
+    if (options.category === NFTCategory.ENS && options.search && db) {
```

Comment:
Why do ENS require special treatment?

Reply:
I added a comment but I forgot to push it 🤦 .
The new search bar needs support for [trigram matching ](https://www.postgresql.org/docs/current/pgtrgm.html) but the graph doesn't support it. In order to accomplish it, I added the logic to first look for the IDs of the ENS that have have more similarity than the threshold and then pass those IDs down to the GraphQL query (for satsuma / thegraph).
I think this way it was easier than having another endpoint o having to refactor the whole thing to fetch all from the db instead of from a subgraph.

Reply:
I think there is a limit to how many ids can be provided in the query.

I don't know the specific amount, but we should consider that if a result returns many ids, we might need to only use x amount or fail with some expected error.

Still, it might not be necessary if the limit is higher than needed.